packages/medusa/src/services/currency.ts

```diff
@@ -0,0 +1,131 @@
+import { TransactionBaseService } from "../interfaces"
+import { EntityManager } from "typeorm"
+import { CurrencyRepository } from "../repositories/currency"
+import { UpdateCurrencyInput } from "../types/currency"
+import { Currency } from "../models"
+import { MedusaError } from "medusa-core-utils"
+import EventBusService from "./event-bus"
+import { FindConfig, Selector } from "../types/common"
+import { buildQuery } from "../utils"
+import { FlagRouter } from "../utils/flag-router"
+import TaxInclusivePricingFeatureFlag from "../loaders/feature-flags/tax-inclusive-pricing"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  currencyRepository: typeof CurrencyRepository
+  eventBusService: EventBusService
+  featureFlagRouter: FlagRouter
+}
+
+export default class CurrencyService extends TransactionBaseService<CurrencyService> {
+  static readonly EVENTS = {
+    UPDATE: "currency.update",
+  }
+
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected readonly currencyRepository_: typeof CurrencyRepository
+  protected readonly eventBusService_: EventBusService
+  protected readonly featureFlagRouter_: FlagRouter
+
+  constructor({
+    manager,
+    currencyRepository,
+    eventBusService,
+    featureFlagRouter,
+  }: InjectedDependencies) {
+    super({ manager })
+    this.manager_ = manager
+    this.currencyRepository_ = currencyRepository
+    this.eventBusService_ = eventBusService
+    this.featureFlagRouter_ = featureFlagRouter
+  }
+
+  /**
+   * Return the currency
+   * @param code - The code of the currency that must be retrieve
+   * @return The currency
+   */
+  async retrieve(code: string): Promise<Currency | never> {
+    const currencyRepo = this.manager_.getCustomRepository(
+      this.currencyRepository_
+    )
+
+    const currency = await currencyRepo.findOne({
+      where: { code },
+    })
```

Comment:
thought: to make the retrieve API easier to work with would it makes sense to normalize the code?

```
const code = rawCode.toLowerCase()
```

Reply:
a lot yeah

Reply:
Also, would it make sense to rename the method to `retrieveByCode` to differentiate it from "normal" `retrieve` methods that always expect an `id`?

Reply:
good idea 👍